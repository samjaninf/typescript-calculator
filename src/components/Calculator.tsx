import { StyleAttribute, css } from 'glamor'
import { h } from 'preact'

type CalculatorProps = {
  children?: JSX.Element,
}

const calculator: StyleAttribute = css({
  background: '#838383',
  borderRadius: 5,
  overflow: 'hidden',
  width: '14em',
})

const Calculator = ({ children }: CalculatorProps): JSX.Element => (
  <div className={calculator.toString()}>
    {children}
  </div>
)

export default Calculator
