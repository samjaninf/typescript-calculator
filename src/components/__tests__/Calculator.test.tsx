import { h } from 'preact'
import Calculator from '../Calculator'
import snapshot from './utils/snapshot'

describe('Calculator', () => {
  it('renders', () => snapshot(Calculator))
  it('renders children', () => snapshot(Calculator, { children: <span /> }))
})
