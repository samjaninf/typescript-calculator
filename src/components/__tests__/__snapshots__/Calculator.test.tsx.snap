// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders 1`] = `"<div class=\\"css-zzci37\\"></div>"`;

exports[`Calculator renders children 1`] = `
"<div class=\\"css-zzci37\\">
  <span></span>
</div>"
`;
